src/dmd/errors.d

```diff
@@ -409,7 +409,10 @@ private void verrorPrint(const ref Loc loc, Color headerColor, const(char)* head
                         dchar u;
                         const msg = utf_decodeChar(line, c, u);
                         assert(msg is null, msg);
-                        fputc(' ', stderr);
+                        // Account for tabs by copying them from the source line
+                        // otherwise cursor will not align correctly when source line contains tabs
+                        // https://issues.dlang.org/show_bug.cgi?id=22678
+                        fputc(u == '\t' ? '\t' : ' ', stderr);
```

Comment:
This is a bit of a wat for me, I'll need a bit longer to think about this.

Reply:
This line emits tab whenever source line has tab, otherwise it prints space. 
```d
// So for source line
tttccccccccccccccccc
// we print
tttssssssssssss^
// t is tab, c is non-tab, s is space
```
This way the cursor is correct for any size of tab char.